tests/forms_tests/widget_tests/test_select.py

```diff
@@ -354,6 +354,86 @@ def test_optgroups_integer_choices(self):
         label, options, index = groups[0]
         self.assertEqual(options[0]['value'], 0)
 
+    def test_optgroups_grouping_of_nogroup_choices(self):
```

Comment:
Is there any reason we can't just add an extra case to the existing `test_optgroups` method above, and so exercise this behaviour, without needing to add this extra method (which is largely identical)?

Reply:
I also thought of adding an extra case to the existing test_optgroups method first. But when I searched on the internet, I found that many people are recommending to add a new test case. 

Maybe I misunderstood (adding a new test case == declaring a new function to reproduce the issue)

We can absolutely add another case in the test_optgroups method.